resotolib/resotolib/graph/__init__.py

```diff
@@ -330,13 +330,16 @@ def search_first_parent_class(self, node, cls):
         This is being used to search up the graph and e.g. find the account that the
         graph node is a member of.
         """
-        queue = deque()
-        queue.extend(self.predecessors(node))
+        queue = deque(self.predecessors(node))
+        already_checked = set()
```

Comment:
WDYT about this? Our implicit contract here is that add the new nodes and mark them as visited in a lockstep, and we want to do the same in the initialization step too.

```suggestion
        queue = deque(set(self.predecessors(node)))
        already_checked = set(self.predecessors(node))
```